backend/api/professor/grading.py

```
@@ -0,0 +1,582 @@
+    
+    cohort = db.query(Cohort).filter(
+        Cohort.id == cohort_assignment.cohort_id,
+        Cohort.created_by == current_user.id
+    ).first()
+    
+    if not cohort:
+        raise HTTPException(status_code=403, detail="Not authorized to grade this submission")
+    
+    # Save previous status for history
+    previous_status = instance.grade_status or "not_graded"
+    previous_grade = instance.grade
```

Ruff (0.14.3)

Local variable `previous_grade` is assigned to but never used

Remove assignment to unused variable `previous_grade`

(F841)